bust-a-move-framework/src/main/java/bustamove/bubble/BubbleStorage.java

```diff
@@ -191,18 +195,19 @@ public boolean isFull() {
      */
     public void removeBubble(final Bubble bubble) {
         for (Bubble[] row : bubble2DArray) {
-            boolean finished = false;
-            if (finished) {
-                break;
-            }
+            boolean stop = false;
             for (int i = 0; i < row.length; i++) {
                 if (row[i] == bubble) {
                     row[i] = null;
-                    finished = true;
+                    stop = true;
                 }
             }
+            if (stop) {

```

Comment:
That is a neater way to do it, but this works as well. It continues to iterate over the innner for loop after the bubble has been removed, but after that it goes inside the conditional and breaks the outer loop.